src/bsp/platform/stm32f4/xi_bsp_io_net_stm32f4.c

```diff
@@ -57,8 +57,8 @@ xi_bsp_io_net_connect( xi_bsp_socket_t* xi_socket, const char* host, uint16_t po
                                    *( ( struct in_addr* )hostinfo->h_addr_list[0] ),
                                .sin_zero = {0}};
 
-    if ( -1 ==
-         connect( *xi_socket, ( struct sockaddr* )&name, sizeof( struct sockaddr ) ) )
+    int ret = connect( *xi_socket, ( struct sockaddr* )&name, sizeof( struct sockaddr ) );
+    if ( -1 == ret )
```

Comment:
No, we don't. This is the remaining part out of my debug session. Will restore the previous implementation.